app/main.py

```
@@ -1,2 +1,117 @@
 class Dictionary:
-    pass
+
+    def __init__(self):
+        self.capacity = 8
+        self.threshold = int(self.capacity * (2 / 3))
+        self.length = 0
+        self.base_data = [None for i in range(self.capacity)]
+
+    def __setitem__(self, key, value):
+        a = 0
+        if self.length >= self.threshold:
+            self.resize()
+
+        index_ = int(hash(key) % self.capacity)
+        if self.base_data[index_] is None:
+            self.base_data[index_] = [key, hash(key), value]
+
+        elif self.base_data[index_][0] == key:
+            self.base_data[index_] = [key, hash(key), value]
+            self.length -= 1
+        else:
+
+            while self.base_data[index_] is not None:
+
+                if self.base_data[index_][0] == key:
+                    self.base_data[index_] = [key, hash(key), value]
+                    self.length -= 1
+                    a = 1
+                if index_ == self.capacity - 1:
+                    index_ = -1
+
+                index_ += 1
+            if a != 1:
+                self.base_data[index_] = [key, hash(key), value]
+        self.length += 1
+
+    def __getitem__(self, key):
+        if self.length != 0:
+            index_ = hash(key) % self.capacity
+            while self.base_data[index_] is not None\
+                    and (index_ < self.capacity):
+                if self.base_data[index_][0] == key:
+                    return self.base_data[index_][2]
+
+                index_ += 1
+
+            raise KeyError(f"Not Key {key} ")
+        else:
+            raise KeyError(f"Not Key {key} ")
```

you should raise an error after while loop